hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/authority/CertificateStore.java

```diff
@@ -74,40 +66,8 @@ void storeValidScmCertificate(BigInteger serialID,
    */
   void checkValidCertID(BigInteger serialID) throws IOException;
 
-
-  /**
-   * Adds the certificates to be revoked to a new CRL and moves all the
-   * certificates in a transactional manner from valid certificate to
-   * revoked certificate state. Returns an empty {@code Optional} instance if
-   * the certificates were invalid / not found / already revoked and no CRL
-   * was generated. Otherwise, returns the newly generated CRL sequence ID.
-   * @param serialIDs - List of Serial IDs of Certificates to be revoked.
-   * @param caCertificateHolder - X509 Certificate Holder of the CA.
-   * @param reason - CRLReason for revocation.
-   * @param revocationTime - Revocation Time for the certificates.
-   * @param approver - CRL approver to sign the CRL.
-   * @return An empty {@code Optional} instance if no CRL was generated.
-   * Otherwise, returns the newly generated CRL sequence ID.
-   * @throws IOException - on failure.
-   */
-  @Replicate
-  Optional<Long> revokeCertificates(List<BigInteger> serialIDs,
-                                    X509CertificateHolder caCertificateHolder,
-                                    CRLReason reason,
-                                    Date revocationTime,
-                                    CRLApprover approver)
-      throws IOException;
-
-  /**
-   * Deletes an expired certificate from the store. Please note: We don't
-   * remove revoked certificates, we need that information to generate the
-   * CRLs.
-   * @param serialID - Certificate ID.
-   */
-  void removeExpiredCertificate(BigInteger serialID) throws IOException;
```

Comment:
Why did you remove this too? Also you didn't remove `removeAllExpiredCertificates` below.

Reply:
This method is defined, and it has two implementation, one in MockCAStore, and SCMCertStore, neither of them did anything meaningful, and actually removing an expired certificate by ID does not seem to be useful even in the future. It is not closely related to CRL, but I ran into this while I was wondering around the code looking at what to remove for CRL.